default.nix

```diff
@@ -1,10 +1,76 @@
-  json-to-haskell = miso.pkgs.haskell.packages.ghcjs.callPackage "/Users/cpenner/dev/json-to-haskell-web/json-to-haskell.nix" {};
-  aeson-extra = miso.pkgs.haskell.packages.ghcjs.callPackage "/Users/cpenner/dev/json-to-haskell-web/aeson-extra.nix" {};
-}
+  };
+
+  miso = import miso-src { };
+
+  # This is just a shortcut for accessing the library of Haskell-related
+  # functions provided by nixpkgs.
+  haskell-lib = miso.pkgs.haskell.lib;
+
+  # This is a set of overrides for the Haskell package set we use to compile
+  # miso. We can define new packages in this set, as well as override/modify
+  # existing packages.
+  my-ghcjs-pkg-set-overrides = final: prev: {
+
+    # Our local app.
+    app = prev.callCabal2nix "app" ./. {};
+
+    # json-to-haskell isn't in the set of Hackage packages that the nixpkgs
+    # from miso knows about, so we can just get it directly from Hackage.
+    # This is generally easier than manually calling `cabal2nix` and storing
+    # the file locally.
+    #
+    # If miso gets updated to provide a more recent version of nixpkgs, then
+    # this can be changed to use the `callHackage` function instead of
+    # `callHackageDirect`.
+    json-to-haskell = final.callHackageDirect {
+      pkg = "json-to-haskell";
+      ver = "0.0.1.0";
+      sha256 = "0a9v8cmcidpav62b1cks41q4s62s34p60ad8isn8kgi34s331xwf";
+    } {};
+
+    # There are various dependencies that appear to be marked as broken.  Here,
+    # I manually mark them as unbroken.  Let's hope that they do actually
+    # compile.
+    aeson-extra = haskell-lib.overrideCabal prev.aeson-extra { broken = false; };
+    time-parsers = haskell-lib.overrideCabal prev.time-parsers { broken = false; };
+
+    # Nonempty-containers depends on the hedgehog-fn package for tests, but it
+    # doesn't have required dependencies.  The easiest fix is just to disable
+    # tests for nonempty-containers.
+    #
+    # Also, haddock generation seems messed up for some reason, so disable that
+    # too.
+    nonempty-containers =
+      haskell-lib.overrideCabal
+        (haskell-lib.dontHaddock (haskell-lib.dontCheck prev.nonempty-containers))
+        { broken = false; };
+
+    # Test suits for various packages don't work when run with GHCJS.  We just
+    # disable these.
+    QuickCheck = haskell-lib.dontCheck prev.QuickCheck;
+    tasty-quickcheck = haskell-lib.dontCheck prev.tasty-quickcheck;
+    scientific = haskell-lib.dontCheck prev.scientific;
+    base-compat-batteries = haskell-lib.dontCheck prev.base-compat-batteries;
+
+    # A bunch of things use doctest in their tests, but doctest doesn't appear
+    # to be able to be compiled, so we just disable tests for all these.
+    comonad = haskell-lib.dontCheck prev.comonad;
+    lens = haskell-lib.dontCheck prev.lens;
+    semigroupoids = haskell-lib.dontCheck prev.semigroupoids;
+    http-types = haskell-lib.dontCheck prev.http-types;
+    servant = haskell-lib.dontCheck prev.servant;
```

Comment:
It seems strange that the haskell package set that `miso` provides doesn't have these types of packages already fixed up.

It seems like something is probably setup incorrectly here.

For instance, it is hard to believe that `miso` doesn't have `QuickCheck` in it's transitive dependency list.